### DISCUSSION TITLE
Control 'Knob' / Radial Control Node / Spin Round Thing with Mouse input (but green)

### DISCUSSION THREAD

**Discussion by yankscally** [godotengine/godot-proposals#6654] [GUI]
A circular rounded knob that is used for typically music related interfaces like volume, and changing the station on a radio.

As Godot has MIDI input it would be nice to have a node for this too. I'm sure that it could have interesting uses as opposed to sliders in games. 

I have already started an assetlib for this node but it feels like it could just be an actual built-in node.

**Comment by Calinou** [comment]
Games rarely use this kind of skeuomorphic knob, as they tend to have poor usability with a mouse. As long as you use `linear_to_db()` when setting the volume and it's wide/tall enough, a traditional slider provides good usability with any input device.

(I'm reminded of [that trend](https://knowyourmeme.com/memes/purposefully-bad-volume-sliders) :stuck_out_tongue:)

**Comment by yankscally** [comment]
How can I represent knobs in a car stereo UI in 2D? What about a control node for safecracking/lockpick puzzles? What about a ship wheel control node? 

Maybe 'radial' control node would imply something more functional and diverse than 'knob node'.

Anything round and you need to spin with the mouse becomes quite heavy in gdscript, and honestly I'm struggling a bit trying to make my assetlib to be nice and jank free.

I understand that sliders are more accurate for mouse movement and are desirable from a design point of view. 
I think that knobs have a more immersive effect for certain controls.

Games rarely tend to have MIDI input, yet there seems to be MIDI messages enabled in Godot, but no way to represent that if I'm using that kind of device. 

Why have MIDI messages in godot if they are rarely used in games? 

While sliders can be replaced with spinboxes, what if you needed a type of slider that took up the least amount of pixels? 

I use these type of control UI when I'm working on music productions and I have no problem with them, I much prefer them over sliders and if they all got replaced with sliders I'd be having a bad day trying to quickly figure out a lot of values by eye.

**Comment by AThousandShips** [comment]
This is a very specific thing that needs to be varied to the specific project, making a design of this that works for everyone isn't easy, and isn't very useful if many end up designing their own implementation anyway because the built in one doesn't fit
